src/Orchard.Cms.Web/Modules/Orchard.Tenants/Controllers/AdminController.cs

```diff
@@ -0,0 +1,359 @@
+
+            S = stringLocalizer;
+            H = htmlLocalizer;
+        }
+
+        public IStringLocalizer S { get; set; }
+        public IHtmlLocalizer H { get; set; }
+
+        public IActionResult Index()
+        {
+            var shells = GetShells();
+
+            var model = new AdminIndexViewModel
+            {
+                ShellSettingsEntries = shells.Select(x => new ShellSettingsEntry { Name = x.Settings.Name, ShellSettings = x.Settings }).ToList()
+            };
+
+            return View(model);
+        }
+
+
+        public async Task<IActionResult> Create()
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants))
+            {
+                return Unauthorized();
+            }
+
+            if (_currentShellSettings.Name != ShellHelper.DefaultShellName)
+            {
+                return Unauthorized();
+            }
+
+            var model = new EditTenantViewModel();
+
+            return View(model);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Create(EditTenantViewModel model)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants))
+            {
+                return Unauthorized();
+            }
+
+            if (_currentShellSettings.Name != ShellHelper.DefaultShellName)
+            {
+                return Unauthorized();
+            }
+
+            if (ModelState.IsValid)
+            {
+                ValidateViewModel(model, true);
+            }
+
+            if (ModelState.IsValid)
+            {
+                var shellSettings = new ShellSettings
+                {
+                    Name = model.Name,
+                    RequestUrlPrefix = model.RequestUrlPrefix,
+                    RequestUrlHost = model.RequestUrlHost,
+                    ConnectionString = model.ConnectionString,
+                    TablePrefix = model.TablePrefix,
+                    DatabaseProvider = model.DatabaseProvider,
+                    State = TenantState.Uninitialized
+                };
+
+                _shellSettingsManager.SaveSettings(shellSettings);
+                var shellContext = _orchardHost.GetOrCreateShellContext(shellSettings);
+
+                return RedirectToAction(nameof(Index));
+            }
+
+            // If we got this far, something failed, redisplay form
+            return View(model);
+        }
+
+        public async Task<IActionResult> Edit(string id)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants))
+            {
+                return Unauthorized();
+            }
+
+            if (_currentShellSettings.Name != ShellHelper.DefaultShellName)
+            {
+                return Unauthorized();
+            }
+
+            var shellContext = GetShells()
+                .Where(x => String.Equals(x.Settings.Name, id, StringComparison.OrdinalIgnoreCase))
```

Comment:
It doesn't help a lot, makes things harder when I want to order them for display.